addons/binding/org.openhab.binding.wifiled/src/main/java/org/openhab/binding/wifiled/handler/LEDStateDTO.java

```diff
@@ -70,18 +75,18 @@ public static LEDStateDTO valueOf(int state, int program, int programSpeed, int
         OnOffType power = (state & 0x01) != 0 ? OnOffType.ON : OnOffType.OFF;
         float[] hsv = new float[3];
         Color.RGBtoHSB(red, green, blue, hsv);
-        long hue = (long) (hsv[0] * 360);
-        int saturation = (int) (hsv[1] * 100);
-        int brightness = (int) (hsv[2] * 100);
+        BigDecimal hue = BigDecimal.valueOf(hsv[0] * 360);
```

Comment:
I am trying to be consistent with the rest of the code.
Since saturation and brightness also are BigDecimals.
I realise that DecimalType does accept a double, however PercentType does not.

What I want to do is get rid of the integer calculations...